e2e/cypress/integration/messaging/long_draft_spec.js

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    before(() => {
+        // # Make sure the viewport is the expected one, so written lines always create new lines
+        cy.viewport(1000, 660);
+
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18699-Leave a long draft in the main input box', () => {
+        const lines = ['Lorem ipsum dolor sit amet,',
+            '\nconsectetur adipiscing elit.',
+            '\nNulla ac consectetur quam.',
+            '\nPhasellus libero lorem,',
+            '\nfacilisis in purus sed, auctor.'];
+
+        // # Create new DM channel with user's email
+        cy.apiGetUsers(['user-1', 'sysadmin']).then((userResponse) => {
+            const userEmailArray = [userResponse.body[1].id, userResponse.body[0].id];
+            cy.wrap(lines.reduce((prev, cur) => prev + cur, '')).as('fullText');
+
+            cy.apiCreateDirectChannel(userEmailArray).then(() => {
+                cy.visit('/ad-1/messages/@sysadmin');
+
+                // # Get the height before starting to write
+                cy.get('#post_textbox', {timeout: TIMEOUTS.LARGE}).type('a').clear();
+                cy.get('#post_textbox').then((post) => {
+                    cy.wrap(parseInt(post[0].clientHeight, 10)).as('previousHeight');
+                });
```

Comment:
Instead of adding explicit wait, you may check its visibility before doing next step.  Above can be combined like:
```javascript
cy.get('#post_textbox').should('be.visible').clear().then((post) => {
    cy.wrap(parseInt(post[0].clientHeight, 10)).as('initialHeight').as('previousHeight');
});
```

Note: I added `initialHeight` alias since it can be used to compare textbox size when switching to a different channel.